UQL/Interpreter.php

```diff
@@ -331,4 +316,79 @@ private function matchDataSourceElement($identifier)
 
         return $this->dataSourceElements[$identifier];
     }
+
+    /**
+     * @param ASTAssertion $astSubtree
+     *
+     * @return array|mixed
+     *
+     * @throws UQLInterpreterException
+     */
+    private function getValue(ASTAssertion $astSubtree)
+    {
+        if ($astSubtree->getValue() instanceof ASTFunctionCall) {
+            return $this->callFunction($astSubtree->getValue());
+        }
+
+        if ($astSubtree->getOperator() == 'T_OP_IN') {
+            if (!($astSubtree->getValue() instanceof ASTArray)) {
+                throw new UQLInterpreterException('Only arrays are valid arguments for IN statements');
+            }
+
+            return $this->parseArray($astSubtree->getValue()->getElements());
+        }
+
+        return $this->parseValue($astSubtree->getValue());
+    }
+
+    /**
+     * @param ASTAssertion $astSubtree
+     *
+     * @return FilterCondition
+     *
+     * @throws UQLInterpreterException
+     */
+    private function getFilterCondition(ASTAssertion $astSubtree)
+    {
+        $dataSourceElement = $this->matchDataSourceElement($astSubtree->getIdentifier());
+        $method = $this->translateOperator($astSubtree->getOperator(), $dataSourceElement);
+        $value = $this->getValue($astSubtree);
+
+        $valueInDatabase = $this->workoutDatabaseValue($value, $dataSourceElement);
+
+        return new FilterCondition($dataSourceElement->getUniqueName(), $method, $value, $valueInDatabase);
+    }
+
+    /**
+     * @param string $value
+     * @param Field  $dataSourceElement
+     *
+     * @return string
+     */
+    private function workoutDatabaseValue($value, Field $dataSourceElement)
+    {
+        $valueInDatabase = $value;
+
+        $dataType = $dataSourceElement->getDataType();
+        if ($dataType instanceof PercentDataType) {
```

Comment:
This looks like something that can be handled better by inheritance. I'm not sure, maybe in Field class.